include/mbedtls/psa_util.h

```diff
@@ -175,8 +180,46 @@ static inline mbedtls_md_type_t mbedtls_md_type_from_psa_alg(psa_algorithm_t psa
 {
     return (mbedtls_md_type_t) (psa_alg & PSA_ALG_HASH_MASK);
 }
+#endif /* MBEDTLS_PSA_CRYPTO_C */
+
+#if defined(MBEDTLS_PSA_UTIL_HAVE_ECDSA)
+
+/** Convert an ECDSA signature from raw format (used by PSA APIs) to DER ASN.1
+ * format (used by legacy crypto APIs).
+ *
+ * \param       raw         Buffer that contains the signature in raw format.
+ * \param       raw_len     Length of raw buffer in bytes
+ * \param[out]  der         Buffer that will be filled with the converted DER
+ *                          output. It can overlap with raw buffer.
+ * \param       der_size    Size of the output der buffer in bytes.
```

Comment:
>  In such a case, there is a natural expectation that the function will succeed even with a smaller output buffer. I think we don't guarantee success in that case

Looking at the current implementation of `mbedtls_ecdsa_raw_to_der` I'm not totally sure we don't support this. There is no "a priori" check on `der_size` and DER writing only fails if the current data (integer, tag or len) cannot be written to the output buffer due to limited size available. Am I wrong?

Reply:
I think that should be easy to test with `ecdsa_raw_to_der_incremental()` by passing input that will result in a smaller DER output and checking that the limit between "fails" and "succeeds" is as expected.

Reply:
Update#2 - In my previous reply I didn't take `convert_raw_to_der_single_int()` into account. Of course you were right in the [above comment](https://github.com/Mbed-TLS/mbedtls/pull/8703#discussion_r1474846583).
Now that you let me notice this I think it would be nice to support also DER buffers that are smaller than the max size for the provided curve's bit-size, so I will update the implementation to reflect this.